File: 2026POTD/JANUARY/26-01-2026/1200.ex
defmodule Solution do
  @spec minimum_abs_difference(arr :: [integer]) :: [[integer]]
  def minimum_abs_difference(arr) do
    a = Enum.sort(arr)
    mn =
      a
      |> Enum.chunk_every(2, 1, :discard)
      |> Enum.map(fn [x, y] -> y - x end)
      |> Enum.min()

    a
    |> Enum.chunk_every(2, 1, :discard)
    |> Enum.filter(fn [x, y] -> y - x == mn end)
  end
end

File: 2026POTD/JANUARY/26-01-2026/1200.go
func minimumAbsDifference(arr []int) [][]int {
    sort.Ints(arr)
    mn := math.MaxInt32
    for i := 1; i < len(arr); i++ {
        if arr[i]-arr[i-1] < mn {
            mn = arr[i] - arr[i-1]
        }
    }
    res := [][]int{}
    for i := 1; i < len(arr); i++ {
        if arr[i]-arr[i-1] == mn {
            res = append(res, []int{arr[i-1], arr[i]})
        }
    }
    return res
}